_episodes/l1-05-continuous_integration.md

```diff
@@ -24,32 +24,468 @@ keypoints:
+> and structure of this YAML file as we go.
+{: .callout}
+
+Let's breakdown the example workflow file in a bit more detail:
+
+```yaml
+on:
+  - push
+```
+
+This describes the condition that will trigger the workflow to run. To add another
+trigger you would add another indented line with a dash, e.g.
+
+```yaml
+on:
+  - push
+  - pull_request
+```
+
+This will additionally trigger the workflow to run when a pull request is created. The
+"push" and "pull_request" triggers are probably the most commonly used however, there
+are a great many available (see [GitHub Docs: Events that rigger
+workflows][actions-triggers]). This is an example of where GitHub Actions goes further
+than most CI systems as you can automate pretty much any behaviour in a repository.
+
+[actions-triggers]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#available-events
+
+Next chunk:
+
+```yaml
+jobs:
+  check-code:
+    runs-on: ubuntu-latest
+    steps:
+```
+
+Workflows are composed of jobs that in turn are composed of steps. In the above we
+create a job called `check-code` by creating an entry in the `jobs` section. We specify
+that the job should run on the most recent version available of Ubuntu (a flavour of
+Linux). Then we go on to define the steps within the job. To add another job to this
+workflow we would add another entry with the same indentation as `check-code` with a
+different name and its own `runs-on` entry and `steps`.
+
+You can read more about jobs at [GitHub Docs: Using jobs in a workflow][jobs]. The
+behaviour of jobs can be extensively modified. You can create dependencies between jobs
+so that `job-2` will only run if `job-1` finished successfully. Or you can provide
+additional expressions to limit when a job should run e.g. only run a job for a
+particular branch. You can also define a single job that is run multiple times with
+different parameters using a matrix. This can be used, for instance, to test code on
+multiple different operating systems with different versions of Python. See [GitHub
+Docs: Using a matrix for your jobs][matrix] for more information.
+
+[matrix]: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs
+[jobs]: https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow
+
+```yaml
+    steps:
+      - uses: actions/checkout@v3
+      - run: echo 'hello world'
+```
+
+Individual steps within a job define the actual work to be carried out. The workflow
+above defines two steps that work in different ways. The first step has a `uses` entry
+to indicate that it should use a pre-packaged action. This is a powerful feature of
+GitHub Actions; individual job steps can be packaged and shared for use in workflows in
+different repositories. Actions that have been packaged this way can be found in the
+[GitHub MarketPlace]. Here we're using version 3 of the `checkout` action which is
```

Comment:
Do we need a link here for `GitHub MarketPlace`?

Reply:
It's defined below - L324